packages/react/src/analytics/integrations/Castle/Castle.ts

```diff
@@ -0,0 +1,328 @@
+/**
+ * Castle Integration.
+ * Tracks page views and user login/logout for security purposes.
+ *
+ * @example <caption>Adding Castle integration to analytics</caption>
+ *
+ * import analytics, \{ integrations \} from '\@farfetch/blackout-react/analytics';
+ *
+ * analytics.addIntegration('castle', integrations.Castle, \{
+ *   pk: \<castle publishable key\>,
+ *   window: \<Overrides the browser window object\>,
+ *   avoidCookies: \<Disables the usage of cookies whenever possible when set to true\>,
+ *   cookieDomain: \<When cookies are used, set the cookie domain scope\>,
+ *   timeout: \<Request timeout for page, form, custom events\>
+ *
+ * \});
+ */
+
+import * as castleJS from '@castleio/castle-js';
+import {
+  eventTypes,
+  LoadIntegrationEventData,
+  StrippedDownAnalytics,
+  trackTypes,
+  UserData,
+  UserTraits,
+  utils,
+} from '@farfetch/blackout-analytics';
+import { Integration } from '@farfetch/blackout-analytics/integrations';
+import coreClient from '@farfetch/blackout-client/helpers/client';
+import identity from 'lodash/identity';
+import isString from 'lodash/isString';
+import pickBy from 'lodash/pickBy';
+import type { AxiosInstance, AxiosRequestConfig } from 'axios';
+import type { CastleInstance, CastleIntegrationOptions } from './types';
+import type {
+  ConfigureOptions,
+  CustomParams,
+  FormParams,
+  PageParams,
+  Thenable,
+  UserParams,
+} from '@castleio/castle-js';
+import type { WebEventData } from '../../types/analytics.types';
+
+export const CLIENT_ID_HEADER_NAME = 'X-Castle-Request-Token';
+export const CASTLE_MESSAGE_PREFIX = 'Castle 2.x -';
+
+/**
+ * Castle integration.
+ */
+class Castle extends Integration<CastleIntegrationOptions> {
+  httpClientInterceptor?: number;
+  isInterceptorAttached?: boolean;
+  debugModeOn: boolean;
+  castleJS: CastleInstance;
```

Comment:
The `CastleInstance` type can be removed and replaced by this which is more type-safe:

```js
castleJS: typeof castleJS
```